cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java

```diff
@@ -239,4 +239,25 @@ public void replicatorBuilderAddsCookieInterceptorDefaultPort() throws Exception
        assertCookieInterceptorPresent(p, "name=%F0%9F%8D%B6&password=%F0%9F%8D%B6");
     }
 
+    @Test
+    public void replicatorBuilderAddsCookieInterceptorSpecialCreds() throws Exception {
+        String encodedUsername = "user%3B%2F%3F%3A%40%3D%26%3C%3E%23%25%7B%7D%7C%5C%5E%7E%5B%5D" +
```

Comment:
I extracted this into a constant with some additional characters and a comment explaining the characters and what the encoded form looks like in 7bd7dfa.